src/service/resource.js

```diff
@@ -334,6 +335,18 @@ export class Resource {
   }
 
   /**
+   * Completes collapse: ensures that the element is `display:none` and
+   * updates layout box.
+   */
+  completeCollapse() {
+    toggle(this.element, false);
+    this.layoutBox_.width = 0;
```

Comment:
Isn't the layout box immutable? How about setting a new `{ width: 0, height: 0 }`?